### PR TITLE
utils: add calls to find host and snap tools

### PR DESCRIPTION
**Pull request by cmatsuoka**
Locate the given tool in the host system or inside a snap. The new
utilities will be used during ELF file handling.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint`?
- [ ] Have you successfully run `pytest tests/unit`?

-----
CRAFT-1189

### CODE REVIEW

File: snapcraft/utils.py [hunk 1]
```diff
@@ -296,13 +297,75 @@ def get_ld_library_paths(prime_dir: Path, arch_triplet: str) -> str:
...
+    if not tool:
+        raise errors.SnapcraftError(
+            f"A tool snapcraft depends on could not be found: {command_name!r}",
+            resolution="Ensure the tool is installed and available, and try again.",
+        )
+    return tool
+
+
+def get_snap_tool(command_name: str) -> str:
+    """Return the path command found in the snap.
```
Comment (mr-cal) [inline snapcraft/utils.py:325]: “Return the path [of a] command found in the snap.”

File: snapcraft/utils.py [hunk 2]
```diff
@@ -296,13 +297,75 @@ def get_ld_library_paths(prime_dir: Path, arch_triplet: str) -> str:
...
+    :return: Path to command
+
+    :raises SnapcraftError: if command_name was not found.
+    """
+    if os.environ.get("SNAP_NAME") != "snapcraft":
+        return get_host_tool(command_name)
+
+    snap_path = os.getenv("SNAP")
+    if snap_path is None:
+        raise RuntimeError("SNAP not defined, but SNAP_NAME is?")
```
Comment (mr-cal) [inline snapcraft/utils.py:340]: Pedantic: “SNAP [environmental variable] not defined,…”

I could see this error being confusing to someone who isn’t thinking about the SNAP environment variable.

File: snapcraft/utils.py [hunk 3]
```diff
@@ -296,13 +297,75 @@ def get_ld_library_paths(prime_dir: Path, arch_triplet: str) -> str:
...
+def _find_command_path_in_root(root: str, command_name: str) -> Optional[str]:
+    for bin_directory in (
+        "usr/local/sbin",
+        "usr/local/bin",
+        "usr/sbin",
+        "usr/bin",
+        "sbin",
+        "bin",
+    ):
+        path = os.path.join(root, bin_directory, command_name)
```
Comment (mr-cal) [inline snapcraft/utils.py:362]: Pedantic: use pathlib

**Review by mr-cal — APPROVED** [review]

File: snapcraft/utils.py (hunk 1 above)
Comment (cmatsuoka) [reply snapcraft/utils.py:325]: Updated.

**Pull request merged**